lm/vocabulary.py

```diff
@@ -1,6 +1,125 @@
-from sisyphus import Job, Task
+from sisyphus import Job, Task, tk
+
+from dataclasses import dataclass
+from typing import Optional, Union
 
 from i6_core.lib.lm import Lm
+from i6_core.lib.lexicon import Lexicon
+from i6_core.util import uopen
+
+
+@dataclass(frozen=True)
+class LmIndexVocabulary:
+    vocab: tk.Path
+    vocab_size: tk.Variable
+    unknown_token: Union[tk.Variable, str]
+
+
+class LmIndexVocabularyFromLexiconJob(Job):
+    """
+    Computes a <word>: <index> vocabulary file from a bliss lexicon for Word-Level LM training
+
+    Sentence begin/end will point to index 0, unknown to index 1.
+    Both are taking directly from the lexicon via the "special" marking:
+      - <lemma special="sentence-begin"> -> index 0
+      - <lemma special="sentence-end"> -> index 0
+      - <lemma special="unknown"> -> index 1
+
+    If <synt> tokens are provided in a lemma, they will be used instead of <orth>
+
+    CAUTION:
+    Be aware of: https://github.com/rwth-i6/returnn/issues/1245 when using Returnn's LmDataset
+    """
+
+    def __init__(
+        self, bliss_lexicon: tk.Path, count_ordering_text: Optional[tk.Path] = None
```

Comment:
pls add docstring. It was not immediately clear to me, what the `count_ordering_text` is used for.